src/citrine/resources/file_link.py

```diff
@@ -0,0 +1,209 @@
+"""A collection of FileLink objects."""
+from uuid import UUID
+import os
+import mimetypes
+from boto3 import client as boto3_client
+from botocore.exceptions import ClientError
+
+from taurus.entity.file_link import FileLink as TaurusFileLink
+from citrine._serialization.properties import String
+from citrine._rest.collection import Collection
+from citrine._rest.resource import Resource
+from citrine._session import Session
+
+
+class _Uploader(object):
+    """Holds the many parameters that are generated and used during file upload."""
+
+    def __init__(self):
```

Comment:
If there is no behavior here, you could probably use a named tuple.

Reply:
That would be cleaner, but named tuples aren't mutable, and the fields are not filled in all at the same time.